unlock-app/src/components/interface/checkout/Connected.tsx

```diff
@@ -1,205 +1,42 @@
-        <DownloadWallet
-          isOpen={isDownloadWallet}
-          setIsOpen={setIsDownloadWallet}
-        />
-        <div className="flex items-center justify-around w-full">
-          <button
-            aria-label="injected wallet"
-            onClick={onInjectedHandler}
-            type="button"
-            className={iconButtonClass}
-          >
-            {ButtonIcon}
-          </button>
-          <button
-            aria-label="wallet connect"
-            onClick={() => authenticateWithProvider('WALLET_CONNECT')}
-            type="button"
-            className={iconButtonClass}
-          >
-            <SvgComponents.WalletConnect width={32} />
-          </button>
-          <button
-            aria-label="coinbase wallet"
-            onClick={() => authenticateWithProvider('COINBASE')}
-            type="button"
-            className={iconButtonClass}
-          >
-            <SvgComponents.CoinbaseWallet width={32} />
-          </button>
-        </div>
-      </div>
-      <div className="flex justify-center col-span-1">
-        <div className="h-full border-l"></div>
-      </div>
-      <div className="grid items-center col-span-5 space-y-2 justify-items-center">
-        <h4 className="text-sm">No crypto wallet?</h4>
-        <Button
-          onClick={(event) => {
-            event.preventDefault()
-            onUnlockAccount()
-          }}
-          size="small"
-          variant="outlined-primary"
-          className="w-full"
-        >
-          Get started
-        </Button>
-      </div>
-    </div>
-  )
-}
+import { Stepper } from './Stepper'
+import { ConnectPage } from './main/ConnectPage'
 
 interface ConnectedCheckoutProps {
-  skipAccountDetails?: boolean
   injectedProvider?: unknown
-  service: CheckoutService | ConnectService
+  service: CheckoutService
   children?: ReactNode
 }
 
 export function Connected({
-  skipAccountDetails = false,
   service,
   injectedProvider,
   children,
 }: ConnectedCheckoutProps) {
   const state = useSelector(service, (state) => state)
-  const { account, email, isUnlockAccount, deAuthenticate, connected } =
-    useAuth()
+  const { account, isUnlockAccount, connected } = useAuth()
   const [signing, setSigning] = useState(false)
+  const { signIn, isSignedIn } = useSIWE()
 
-  const { authenticateWithProvider } = useAuthenticate({
-    injectedProvider,
-  })
-  const { signIn, signOut, isSignedIn } = useSIWE()
-  const [isDisconnecting, setIsDisconnecting] = useState(false)
   const useDelegatedProvider =
     state.context?.paywallConfig?.useDelegatedProvider
 
   useEffect(() => {
     const autoSignIn = async () => {
+      // Skip Connect if already signed in
       if (!isSignedIn && !signing && connected && isUnlockAccount) {
         await signIn()
+        service.send({ type: 'SELECT_LOCK' })
+      } else if (isSignedIn && !signing && isUnlockAccount) {
+        await signIn()
+        service.send({ type: 'SELECT_LOCK' })
```

Comment:
Since these 2 cases are the same can you extract the conditional logic like

```js
const isConnectedAsUnlockAccount = !isSignedIn && !signing && connected && isUnlockAccount

const isConnectedWithWallet = isSignedIn && !signing && isUnlockAccount
```

and then have a single 
```
if(isConnectedAsUnlockAccount || isConnectedWithWallet) {
...
}
```
```